### DISCUSSION TITLE
Cursor shape changes in tmux

### DISCUSSION THREAD

**Discussion by tembbo** [ghostty-org/ghostty#3690] [Q&A] [answered]
I have no option setting the cursor shape in my ghostty config, so it's set as a line/beam. But when I enter tmux it suddenly transitions into a block. Is there anyway to maintain the same cursor shape in tmux?

I have tried setting `set -g cursor-style "bar"` in my tmux.conf, but that only works partially on new sessions where I haven't opened neovim yet, because when i open neovim and then close it, the cursor is back to block

**Comment by 00-kat** [comment]
If you're in the Ghostty Discord server (join with https://discord.gg/ghostty), check out this thread: https://discord.com/channels/1005603569187160125/1278793281228247111/1278793281228247111.

If you're not, please mention your shell, and I'll also copy the text from there over to here.

**Reply by tembbo** [reply]
fish 3.7.1

**Reply by 00-kat** [reply]
Fish since version 4 has a better default cursor shape if you use Vi mode; but of course you don't use that.

Also, Fish didn't have Ghostty in the allow-list (https://github.com/fish-shell/fish-shell/pull/10685), which has been merged now, but isn't in Fish 3.7.1.

(Note: I'm 90% sure that the following works even if you don't use Vi mode, but I don't know how to check)

So add this to your Fish configuration:
```fish
if status is-interactive
    if string match -q -- '*ghostty*' $TERM
        set -g fish_vi_force_cursor 1
    end
end
```

And there's some variables you can use to modify the cursor shape, the documentation for which can be found [here](https://fishshell.com/docs/current/interactive.html#vi-mode-commands). Here's the entire list: `fish_cursor_insert` `fish_cursor_visual` `fish_cursor_default` `fish_cursor_replace` `fish_cursor_unknown` `fish_cursor_external` `fish_cursor_replace_one` `fish_cursor_selection_mode` (if you don't use Vi mode, you're likely to only be interested in `fish_cursor_default` and `fish_cursor_external`)

**Reply by tembbo** [reply, marked as answer]
I switched to vi mode and now it works, thanks anyways
